### PR TITLE
Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by HIGGO36**
#### What's this PR do?

**Upgrade to Babel 7**

Some references:
    https://babeljs.io/docs/en/v7-migration
    https://github.com/babel/babel-upgrade/blob/master/readme.md
    https://blog.bitsrc.io/so-whats-new-in-babel-7-ea97cb984ef0

#### Where should the reviewer start?

Start with reading over upgrade details provided in reference links above.
Then maybe review and compare changes made to /package.json, /package-lock.json, and src/cbp-theme.browser.bundle.js files with current versions.

#### How should this be manually tested?

Test known working ES6 or ES7 code currently being used as deemed necessary.

#### Any background context you want to provide?

While researching a known .app-header functionality issue, we found some syntax errors from the cbp-theme.js file, which led our team to look at the current usage of babel for compiling ES6 to ES5 for browser compatibility using babel-polyfill. 

Next, we found a quick fix by making the polyfill import first before any other dependency imports from the ' /cbp-theme.browser.bundle.js ' .

However, in order to provide a more permanent and scalable fix that would avoid possible conflicts arising from multiple versions of babel-polyfill being imported, we found that Babel 7 provides a fix ( 'Automatic Polyfilling' ) that allows for us to Not need to import the polyfill at the entry point of the app.

In addition, Babel 7 includes many other neat features/additions that we could also benefit from down the road. (ex: Support for TypeScript, faster build times, etc...)

#### What are the relevant tickets?
n/a

#### Screenshots (if appropriate)
n/a

#### Questions:
- Is there a blog post?

#271 

- What other upgrade steps need to be done?

Updates have been tested in a demo build, not seeing any errors or bugs.

- Does the knowledge base need an update?

*Other modules were updates to work well with Babel 7 and help with some other needed fixes for the theme.

Updated to:
- core-js -v 3
- js-yaml -v 3.13.1
- jest-cli -v 24.7.1
- cssnano -v 4.1.10
- braces -v 3.0.1
- rollup-plugin-sass -v 1.2.2
- babel-jest -v 24.7.1

and more.....can see all within the changes noted within the package.json file.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -116,6 +123,8 @@
   },
   "dependencies": {
     "bootstrap-sass": "3.3.7",
```
Comment (davezen1) [inline package.json]: Move braces and core-js to devDependencies

File: package.json [hunk 2]
```diff
@@ -37,19 +41,22 @@
     "node": ">= 8"
   },
   "devDependencies": {
```
[the remark this answers is not in the document]
Comment (HIGGO36) [reply package.json]: gotcha, will make sure to use this method for future installs

File: src/cbp-theme.browser.bundle.js [hunk 3]
```diff
@@ -9,10 +9,10 @@ import 'mdl-selectfield/dist/mdl-selectfield'
 
 import {CBPTheme} from './cbp-theme'
```
[the remark this answers is not in the document]
Comment (HIGGO36) [reply src/cbp-theme.browser.bundle.js]: Ok, removed the commented code out.

File: package.json (hunk 1 above)
Comment (HIGGO36) [reply package.json]: ok thx, moved both braces and core-js to devDependencies.

**Pull request merged**